File: Rea's code/10:13 - 10:15/HouseRobber.java

public class HouseRobber {
	public int rob (int[] nums){
		if (nums.length == 0 || nums == null){
			return 0;
		}
		int max [] = new int [nums.length + 1];
		max [0] = 0;
		max [1] = nums[0];
		for (int i = 2; i < max.length; i ++){
			max[i] = Math.max(max[i -1], max[i - 2] + nums[i -1]);
		}
		return max[max.length -1];
	}
}

File: mu_wang/10_13/House Robber.py
class Solution(object):
    def rob(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        last, now = 0, 0

        for i in nums: last, now = now, max(last + i, now)

        return now

    def rob(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        a = 0
        b = 0
        for idx,i in enumerate(nums):
            if idx % 2==0:
                a = max(a+i,b)
            else:
                b = max(b+i,a)
        return max(a,b)